DevelopGitHub/GitHubExtension/GitHubExtention.Preferences.WebApi/Controllers/UserPreferencesController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Data.Entity;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading.Tasks;
+using System.Web.Http;
+using GitHubExtension.Infrastructure.Constants;
+using GitHubExtension.Preferences.DAL.Model;
+using GitHubExtention.Preferences.WebApi.Constants;
+using GitHubExtention.Preferences.WebApi.Extensions;
+using GitHubExtention.Preferences.WebApi.Models;
+using GitHubExtention.Preferences.WebApi.Queries;
+using Microsoft.AspNet.Identity;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace GitHubExtention.Preferences.WebApi.Controllers
+{
+    [RoutePrefix(RouteConstants.ApiUser)]
+    public class UserPreferencesController : ApiController
+    {
+        private readonly IPreferencesQuery _query;
+
+        private readonly IAzureContainerQuery _container;
+
+        public UserPreferencesController(IPreferencesQuery query, IAzureContainerQuery container)
+        {
+            _query = query;
+            _container = container;
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        [Route(RouteConstants.Avatar)]
+        public async Task<IHttpActionResult> Post(FileModel file)
+        {
+            User user = _query.GetUser(User.Identity.GetUserId());
+            if (user == null)
+            {
+                return NotFound(); 
+            }
+
+            string newFileUrl = _container.UpdateBlob(file);
+            if (IsFileLocationValid(user, newFileUrl))
```

Comment:
Please rename to something like IsUrlValidbecause I don't see that we work with actual files. Only urls. Correct?